ground/src/main/java/com/google/android/ground/persistence/remote/firebase/FirestoreDataStore.kt

```diff
@@ -64,8 +70,14 @@ internal constructor(
   override suspend fun loadTermsOfService(): TermsOfService? =
     withContext(ioDispatcher) { db().termsOfService().terms().get() }
 
-  override suspend fun loadSurveySummaries(user: User): List<Survey> =
-    withContext(ioDispatcher) { db().surveys().getReadable(user) }
+  override fun getSurveyList(user: User): Flow<List<SurveyListItem>> = flow {
+    emitAll(
+      db().surveys().getReadable(user).map { list ->
+        // TODO(#2031): Return SurveyListItem from getReadable(), only fetch required fields.
+        list.map { it.toListItem(false) }
+      }
+    )
+  }
```

Comment:
IIUC, `getReadable()` already returns a flow. Why can't we directly return it with transform instead of using `flow { emitAll() }`?

Reply:
got it